src/main/java/com/sap/oss/phosphor/fosstars/data/github/LocalRepository.java

```diff
@@ -334,6 +341,79 @@ public void close() {
     repository.close();
   }
 
+  /**
+   * Estimate how much the project has changed since a specified data.
+   * The method inspects the commit history after the specified data,
+   * and looks for updated files.
+   * The method doesn't take into account what was exactly updated in the files.
+   *
+   * @param date The date.
+   * @param toConsider Defines which files should be considered.
+   * @return An estimated amount of changes in the project.
+   * @throws IOException If something went wrong.
+   */
+  public Double changedSince(Date date, Predicate<Path> toConsider) throws IOException {
+    loadCommitsIfNecessary();
+
+    GitCommit head = commits.get(0);
```

Comment:
If commits are empty even after trying to load commits and there were no exceptions but still empty. Could this happen?

Reply:
It is better to add a check here.